lib/api/core/indexCache.js

```diff
@@ -52,6 +53,16 @@ class IndexCache {
           promises.push(this.kuzzle.internalEngine.listCollections(index)
             .then(collections => {
               this.indexes[index] = collections;
+
+              const mappingPromises = collections.map(collection => {
+                const mapping = {
+                  properties: this.commonMapping
+                };
+
+                return this.kuzzle.internalEngine.updateMapping(collection, mapping, index);
+              });
+
+              return Promise.all(mappingPromises);
```

Comment:
Always use Bluebird instead of native promises: `return Bluebird.all(mappingPromises);`